sync.js

```diff
@@ -59,7 +88,35 @@ class Sync extends events.EventEmitter {
     }
   }
 
-  start ({host, port}, opts) {
+  replicate (target, opts) {
+    if (!target) throw new Error('target required, got null')
+    if (!opts) opts = {}
+    target.name = target.filename || target.name
+    this._openFeeds[target.name] = OpenConnection(target, REPLICATION_WAITING)
+
+    var emitter
+    if (target.filename) {
+      emitter = this.replicateFromFile(target.filename, opts)
+    } else if (target.host && target.port) {
+      emitter = this.replicateNetwork(target, opts)
+    } else throw new Error('Requires filename or host and port')
+
+    emitter.on('progress', (progress) => {
+      this._openFeeds[target.name] = OpenConnection(target, REPLICATION_PROGRESS, progress)
+    })
+
+    emitter.on('error', (error) => {
+      this._openFeeds[target.name] = OpenConnection(target, REPLICATION_ERROR, error)
+    })
+
+    emitter.on('end', () => {
+      this._openFeeds[target.name] = OpenConnection(target, REPLICATION_COMPLETE)
```

Comment:
Do feeds ever get removed from the `_openFeeds` map?

Reply:
clear() function does this, can rename to clearState to make this more clear

Reply:
Oh, is the idea that the client needs to manually clear all sync state? When does it do that?

Reply:
right now yea but in the future it'd be better to be able to remove the state for a specific peer, for example it lets you know that it's complete and then you hit the 'x' button in the ui and the state should be ignored/removed after that.
also if we want sync history this might become irrelevant. this is a band aid